python/tank/bootstrap/manager.py

```diff
@@ -690,13 +693,27 @@ def get_pipeline_configurations(self, project):
             current_login=self._sg_user.login,
             sg_connection=self._sg_connection,
         ):
-            pcs.append({
+            pipeline_config_data = {
                 "id": pc["id"],
                 "type": pc["type"],
                 "name": pc["code"],
                 "project": pc["project"],
                 "descriptor": pc["config_descriptor"],
-            })
+                "descriptor_uri": None,
```

Comment:
May I suggest `descriptor_source` or `original_descriptor`? `descriptor` points to the resolved version.

Reply:
I also find it confusing that one part of the API, `descriptor`, returns a descriptor object while another, `descriptor_uri`, returns a string.

Reply:
I agree this is a little bit confusing - i will try my best to add comments and naming things to minimize confusion. `descriptor_source` is great - i might go with `descriptor_source_uri` to further hint that it's a string, not an object.

Tests turned out to be really hard to add, i'll take a another look to see if i can wrangle it somehow.